sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_challenge_auth_policy.py

```diff
@@ -15,65 +15,60 @@
 """
 from typing import TYPE_CHECKING
 
-from azure.core.pipeline.policies import AsyncHTTPPolicy
+from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy
 
-from . import HttpChallengeCache
-from .challenge_auth_policy import _enforce_tls, _get_challenge_request, _update_challenge, ChallengeAuthPolicyBase
+from . import http_challenge_cache as ChallengeCache
+from .challenge_auth_policy import _enforce_tls, _update_challenge
 
 if TYPE_CHECKING:
-    from typing import Any
-    from azure.core.credentials_async import AsyncTokenCredential
-    from azure.core.pipeline import PipelineRequest
-    from azure.core.pipeline.transport import AsyncHttpResponse
-    from . import HttpChallenge
+    from typing import Any, Optional
+    from azure.core.pipeline import PipelineRequest, PipelineResponse
 
 
-class AsyncChallengeAuthPolicy(ChallengeAuthPolicyBase, AsyncHTTPPolicy):
+class AsyncChallengeAuthPolicy(AsyncBearerTokenCredentialPolicy):
     """policy for handling HTTP authentication challenges"""
 
-    def __init__(self, credential: "AsyncTokenCredential", **kwargs: "Any") -> None:
-        self._credential = credential
-        super(AsyncChallengeAuthPolicy, self).__init__(**kwargs)
-
-    async def send(self, request: "PipelineRequest") -> "AsyncHttpResponse":
+    def __init__(self, *args: "Any", **kwargs: "Any") -> None:
+        self._last_tenant_id = None  # type: Optional[str]
+        super().__init__(*args, **kwargs)
+
+    async def on_request(self, request: "PipelineRequest") -> None:
+        challenge = ChallengeCache.get_challenge_for_url(request.http_request.url)
+        if challenge:
+            if self._last_tenant_id == challenge.tenant_id:
```

Comment:
When is this False? `on_challenge` updates the cached challenge and sets `_last_tenant_id` to match, so it appears they will always be equal.

Reply:
Hm, yeah I think you're right. I was confusing this with evaluating the tenant ID of a new challenge, but this is being fetched from the cache so they should always be equal.